core/src/main/scala/org/apache/spark/internal/io/SparkHadoopWriter.scala

```diff
@@ -193,11 +193,15 @@ class HadoopMapRedWriteConfigUtil[K, V: ClassTag](conf: SerializableJobConf)
   override def createCommitter(jobId: Int): HadoopMapReduceCommitProtocol = {
     // Update JobConf.
     HadoopRDD.addLocalConfiguration("", 0, 0, 0, getConf)
+    var path = getConf.get("mapred.output.dir")
+    if (path == null) {
+      path = Utils.createTempDir().getPath + "/mapred-output-dir"
+    }
```

Comment:
Doing this in a general case would not be correct : though specifically for SHC, given its semantics, it would be fine.
If we want to support this properly in spark, we would need to modify commit handling in spark to not require output directory ...